interface-definitions/service_config-sync.xml.in

```diff
@@ -73,30 +73,79 @@
               </constraint>
             </properties>
           </leafNode>
-          <leafNode name="section">
+          <node name="section">
             <properties>
               <help>Section for synchronization</help>
-              <completionHelp>
-                <list>nat nat66 firewall</list>
-              </completionHelp>
-              <valueHelp>
-                <format>nat</format>
-                <description>NAT</description>
-              </valueHelp>
-              <valueHelp>
-                <format>nat66</format>
-                <description>NAT66</description>
-              </valueHelp>
-              <valueHelp>
-                <format>firewall</format>
-                <description>firewall</description>
-              </valueHelp>
-              <constraint>
-                <regex>(nat|nat66|firewall)</regex>
-              </constraint>
-              <multi/>
             </properties>
-          </leafNode>
+            <children>
+              <leafNode name="firewall">
+                <properties>
+                  <help>Firewaall</help>
+                  <valueless/>
+                </properties>
+              </leafNode>
+              <tagNode name="interfaces">
+                <properties>
+                  <help>Interfaces</help>
+                </properties>
+                <children>
+                  #include <include/generic-description.xml.i>
+                </children>
+              </tagNode>
+              <leafNode name="nat">
+                <properties>
+                  <help>NAT</help>
+                  <valueless/>
+                </properties>
+              </leafNode>
+              <leafNode name="nat66">
+                <properties>
+                  <help>NAT66</help>
+                  <valueless/>
+                </properties>
+              </leafNode>
+              <leafNode name="pki">
+                <properties>
+                  <help>VyOS PKI configuration</help>
+                  <valueless/>
+                </properties>
+              </leafNode>
+              <leafNode name="policy">
+                <properties>
+                  <help>Routing policy</help>
+                  <valueless/>
+                </properties>
+              </leafNode>
+              <tagNode name="protocols">
+                <properties>
+                  <help>System services</help>
+                </properties>
+                <children>
+                  #include <include/generic-description.xml.i>
+                </children>
+              </tagNode>
+              <tagNode name="service">
+                <properties>
+                  <help>System services</help>
+                </properties>
```

Comment:
It was attempt to add them dynamically without completion.
This way user can add any service.
That’s why I use tagNode without completion/constraint.
Do you think we have to limit services/interfaces?

Reply:
Well you can add any service but what if user adds service `foo` that does not exist at all? Having no control over the input dat is general a bad idea. If there is a completion help and constraint it can be extended super easily thus we should add it to also show the user the possibilities he has.

Reply:
Can we generate completion help/constraint dynamically, dependent on our subsections of the service?

Reply:
Re-implemented